src/katsdpcontroller/generator.py

```diff
@@ -1166,20 +1168,23 @@ def get_baseline_index(a1, a2):
                 # least 1 and that could be larger.
                 mid_vis_size = max(100 * 1024 * 1024, vis_size * 2)
                 xbgpu.mem += _mb(vis_size * 5)  # Magic number is default in XSend class
-                xbgpu.gpus[0].mem += _mb(2 * vis_size + mid_vis_size)
+                xbgpu.gpus[0].compute += 0.15 * bw_scale
+                xbgpu.gpus[0].mem += _mb(n_tx_items * vis_size + mid_vis_size)
                 # Minimum capability as a function of bits-per-sample, based on
                 # tensor_core_correlation_kernel.mako from katgpucbf.xbgpu.
                 min_compute_capability = {4: (7, 3), 8: (7, 2), 16: (7, 0)}
                 xbgpu.gpus[0].min_compute_capability = min_compute_capability[acv.bits_per_sample]
             elif isinstance(stream, product_config.GpucbfTiedArrayChannelisedVoltageStream):
-                # TODO: NGC-1222 Update xbgpu.mem and xbgpu.gpus[0].mem
-                beam_size = (
-                    batches_per_chunk
-                    * stream.n_chans_per_substream
-                    * stream.spectra_per_heap
-                    * COMPLEX
+                elements = (
+                    batches_per_chunk * stream.n_chans_per_substream * stream.spectra_per_heap
                 )
-                xbgpu.mem += _mb(beam_size * 2)  # Magic number default for xbgpu
+                beam_size = elements * COMPLEX
+                rand_state_size = elements * 48  # 48 is sizeof(curandStateXORWOW_t)
+                xbgpu.mem += _mb(beam_size * n_tx_items)
+                # Allow 128 single-pol beams + 1 baseline-correlation-products for 80 antennas.
+                xbgpu.gpus[0].compute += 0.125 / n_inputs * bw_scale
+                xbgpu.gpus[0].mem += _mb(beam_size * n_tx_items)
```

Comment:
If I could be a small, but non-zero pain
* Could I request you update this calc to be in the ~same order as the one for X-engine config?
* i.e. `_mb(n_tx_items * beam_size)`  - "multiplier" first

Reply:
Done.